### DISCUSSION TITLE
Migrate 'generic' to 'generic_esp32'

### DISCUSSION THREAD

**Discussion by FinalD123** [tbnobody/OpenDTU#2315] [Q&A]
Hello,
I have currently installed version [v23.12.18](https://github.com/tbnobody/OpenDTU/releases/tag/v23.12.18) wit PIO environment "generic". Looking at the [firmware installation guide](https://www.opendtu.solar/firmware/#overview-of-firmware-variants) I should migrate to the ESP32 version of the image because the "generic" version is marked as deprecated?
What I'm not sure about is if I should really use the provided device file in step 3 of the migration guilde because my "nrf24" pins are different compared to the default values?

I also read these two discussions (https://github.com/tbnobody/OpenDTU/discussions/1619 and https://github.com/tbnobody/OpenDTU/discussions/1509) but they don't mention anything about different PIN mapping for the "nrf24" module?

Thanks in advance.

**Comment by stefan123t** [comment]
@FinalD123 you can make a backup of your config and pin_mapping.json in case you do not remeber your cabling.
ESP32 is only supported by OpenDTU, there are different ESP32 models like the S3 which may require a special binary. What type of ESP32 do you have ?

**Comment by FinalD123** [comment]
@stefan123t: I have a "ESP32 NodeMCU". If I'm not mistaken I need the standard ESP32 version of the image.

I will defnitely do a backup but maybe we are talking past each others. Following steps are required for a migration:
* Make a backup of your configuration! (Settings --> Config Management --> Backup --> Select config.json --> Press Backup)
* Download the latest opendtu-generic_esp32.bin file from [here](https://github.com/tbnobody/OpenDTU/releases)
* Download [this](https://raw.githubusercontent.com/tbnobody/OpenDTU/master/docs/DeviceProfiles/nodemcu_esp32.json) device profile. It contains the same pin assignments as the hard-coded ones in opendtu-generic.bin.
* Upload the device profile. (Settings --> Config Management --> Restore --> Select Pin Mapping (pin_mapping.json) --> Select nodemcu_esp32.json --> Press Restore)
* The ESP reboots but does nothing different at the moment.
* Select the new uploaded device profile. This can already be done using the opendtu-generic.bin firmware. (Settings --> Device-Manager --> Selected profile --> Select NRF24 --> Press Save)
* The ESP reboots and uses the pin assignment from the device profile. (Which makes no difference because they are equal to the old one)
* Upload the new firmware (Settings --> Firmware Upgrade)
* The ESP reboots

I don't fully understand why should I upload a device profile in the 4th step that seems not matching my current PIN mapping? Should I directly overwrite it with my required mapping configuration after the last step?
